### PR TITLE
add sloppy-deployment

### PR DESCRIPTION
**Pull request by kr1sp1n**
This is the first draft of a generic deployment strategy for sloppy.io.
The idea is to have a project-specific `environments` directory with environment-specific files and a `deployment` directory with a template file `sloppy.tmpl.yml`:
```bash
project
├── deployment
│   ├── Dockerfile
│   └── sloppy.tmpl.yml
├── docker-compose.ops.yml
└── environments
    ├── global.json
    └── test
        ├── domains.yml
        ├── sloppy.env
        └── website.env
```

You can deploy via docker-compose:
```bash
docker-compose -f docker-compose.ops.yml up --force-recreate --build --exit-code-from deploy-test --abort-on-container-exit deploy-test
```
Important is the `deployment/Dockerfile` that just needs one line: ```FROM 7val/sloppy-deployment:latest```. There are some conventions build in and an `ONBUILD` command to make it work.

### CODE REVIEW

File: sloppy-deployment/test/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: '2.3'
```
Comment (malud) [inline sloppy-deployment/test/docker-compose.yml]: The sloppy-cli does support compose-format version 3 only.

File: sloppy-deployment/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -eu
```
Comment (malud) [inline sloppy-deployment/entrypoint.sh]: Which problem is getting solved here? sloppy-cli convert docker-compose.yml files to sloppy.yml via `run` command per default. Environment vars or env files gets evaluated too. For a conversion without api access you could consider the https://github.com/sloppyio/sloppose project.

File: sloppy-deployment/test/docker-compose.yml (hunk 1 above)
Comment (kr1sp1n) [reply sloppy-deployment/test/docker-compose.yml]: This `docker-compose.yml` is for testing only. Not to be meant as parsing config for sloppy CLI.

File: sloppy-deployment/entrypoint.sh (hunk 2 above)
Comment (kr1sp1n) [reply sloppy-deployment/entrypoint.sh]: I did not realize that the CLI is capable of this ;)

Comment (kr1sp1n) [reply sloppy-deployment/entrypoint.sh]: @malud Sometimes we need env-specific data that do not fit into env vars so we need a way to process more complex configs out of files. Like a list of 200 values that must be present in the sloppy config.

**Pull request merged**